### PR TITLE
[flutter_tools] remove mocks from devfs web, cache, and xcode migrator test

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards #71511

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart [hunk 1]
```diff
@@ -41,8 +33,7 @@ void main() {
     );
     await createTestCommandRunner(command).run(const <String>['precache']);
 
-    // Do not throw StateError, lock is acquired.
-    expect(() => cache.checkLockAcquired(), returnsNormally);
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:45]: This was testing a mocked cache, which returned null for this method and would have never failed

File: packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart [hunk 2]
```diff
@@ -61,9 +52,7 @@ void main() {
     );
     await createTestCommandRunner(command).run(const <String>['precache']);
 
-    expect(Cache.isLocked(), isFalse);
-    // Do not throw StateError, acquired reentrantly with FLUTTER_ALREADY_LOCKED.
-    expect(() => cache.checkLockAcquired(), returnsNormally);
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:66]: Same here

File: packages/flutter_tools/test/general.shard/web/devfs_web_test.dart [hunk 3]
```diff
@@ -883,18 +853,6 @@ void main() {
     outputFile.parent.childFile('a.json').writeAsStringSync('{}');
     outputFile.parent.childFile('a.map').writeAsStringSync('{}');
 
-    final ResidentCompiler residentCompiler = MockResidentCompiler();
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/web/devfs_web_test.dart:886]: Not even used!

File: packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart [hunk 4]
```diff
@@ -41,8 +33,7 @@ void main() {
     );
     await createTestCommandRunner(command).run(const <String>['precache']);
 
-    // Do not throw StateError, lock is acquired.
-    expect(() => cache.checkLockAcquired(), returnsNormally);
+    expect(cache.locked, true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:36]: I'm just curious, does this command not release the lock at the end?

Comment (jonahwilliams) [reply packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:36]: It does - this test is just asserting that the command locked it at the start

File: packages/flutter_tools/test/general.shard/web/devfs_web_test.dart [hunk 5]
```diff
@@ -1107,5 +1041,29 @@ void main() {
   }));
 }
 
-class MockHttpServer extends Mock implements HttpServer {}
-class MockResidentCompiler extends Mock implements ResidentCompiler {}
+class MockHttpServer extends Fake implements HttpServer {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/web/devfs_web_test.dart:1044]: Should this be `FakeHttpServer`?

File: packages/flutter_tools/test/general.shard/web/devfs_web_test.dart [hunk 6]
```diff
@@ -1107,5 +1041,29 @@ void main() {
...
+  bool closed = false;
+
+  @override
+  Future<void> close({bool force = false}) async {
+    closed = true;
+  }
+}
+
+class MockResidentCompiler extends Fake implements ResidentCompiler {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/web/devfs_web_test.dart:1053]: Ditto

File: packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart (hunk 4 above)
Comment (christopherfujino) [reply packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:36]: Does `await`-ing `runner.run()` not block until the command exits?

Comment (jonahwilliams) [reply packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:36]: the command never exits in the test scripts

Comment (christopherfujino) [reply packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:36]: TIL

Comment (jonahwilliams) [reply packages/flutter_tools/test/commands.shard/hermetic/precache_test.dart:36]: ahh, actually no - the command acquires the lock but the release is managed by the runner elsewhere during shutdown or Cache.releaseLockEarly

File: packages/flutter_tools/test/general.shard/web/devfs_web_test.dart (hunk 6 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/web/devfs_web_test.dart]: done

File: packages/flutter_tools/test/general.shard/web/devfs_web_test.dart (hunk 5 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/web/devfs_web_test.dart]: done

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**